packages/engine/Source/Scene/Model/PntsLoader.js

```diff
@@ -572,6 +581,473 @@ function makeStructuralMetadata(parsedContent, customAttributeOutput) {
+ * @param point
+ * @param a + t*b
+ * @returns t minimizing |point - (a + t*b)|
+ */
+function closestPointOnLineToPoint(point, a, b) {
+  const dir = normalize(b);
+  return (
+    dir[0] * (point[0] - a[0]) +
+    dir[1] * (point[1] - a[1]) +
+    dir[2] * (point[2] - a[2])
+  );
+}
+
+function closestPointOnRayToPoint(point, a, b) {
+  const t = closestPointOnLineToPoint(point, a, b);
+  if (t < 0.0) {
+    return a;
+  }
+  const intersection = new Float64Array(3);
+  intersection[0] = a[0] + t * b[0];
+  intersection[1] = a[1] + t * b[1];
+  intersection[2] = a[2] + t * b[2];
+  return intersection;
+}
+
+function distanceFromRayToPoint(ray, point) {
+  const origin = new Float64Array(3);
+  const dir = new Float64Array(3);
+  origin[0] = ray.origin.x;
+  origin[1] = ray.origin.y;
+  origin[2] = ray.origin.z;
+  dir[0] = ray.direction.x;
+  dir[1] = ray.direction.y;
+  dir[2] = ray.direction.z;
+  return distance(point, closestPointOnRayToPoint(point, origin, dir));
+}
+
+function boundingSphereOfBox(minx, maxx, miny, maxy, minz, maxz) {
+  const center = new Float64Array(3);
+  center[0] = (minx + maxx) / 2;
+  center[1] = (miny + maxy) / 2;
+  center[2] = (minz + maxz) / 2;
+  const vertex = new Float64Array(3);
+  vertex[0] = minx;
+  vertex[1] = miny;
+  vertex[2] = minz;
+  const radius = distance(center, vertex);
+  return [center, radius];
+}
+
+PntsLoader.prototype.findPointsWithinRadiusOfRay = function (ray, radius) {
+  const positions = this._enuCoords;
+  const indexedTree = this._3DTree;
+  const box = new Float64Array(6);
+  box[0] = this._boxENU[0];
+  box[1] = this._boxENU[1];
+  box[2] = this._boxENU[2];
+  box[3] = this._boxENU[3];
+  box[4] = this._boxENU[4];
+  box[5] = this._boxENU[5];
+  const curNode = 0;
+  const depth = 1;
+  const result = rayIterate(
+    indexedTree,
+    positions,
+    ray,
+    box,
+    curNode,
+    depth,
+    radius
+  );
+  if (!result) {
+    return null;
+  }
+  const ecefTransformationMatrix = Transforms.eastNorthUpToFixedFrame(
+    this._ecefRefPoint
+  );
+  const ecef = matrixMultbyPointasVec(
+    ecefTransformationMatrix,
+    result[0],
+    result[1],
+    result[2]
+  );
+  return new Cartesian3(
+    ecef[0] + this._ecefRefPoint.x,
+    ecef[1] + this._ecefRefPoint.y,
+    ecef[2] + this._ecefRefPoint.z
+  );
+};
+
+function rayIterate(
+  indexedTree,
+  positions,
+  ray,
+  boundingBox,
+  curNode,
+  depth,
+  radius
+) {
+  const curx = positions[curNode * 3];
+  const cury = positions[curNode * 3 + 1];
+  const curz = positions[curNode * 3 + 2];
+  const curPoint = new Float64Array(3);
+  curPoint[0] = curx;
+  curPoint[1] = cury;
+  curPoint[2] = curz;
+  const sphere = boundingSphereOfBox(
+    boundingBox[0],
+    boundingBox[1],
+    boundingBox[2],
+    boundingBox[3],
+    boundingBox[4],
+    boundingBox[5]
+  );
+  if (distanceFromRayToPoint(ray, sphere[0]) > sphere[1] + radius) {
+    return null;
+  }
+  if (distanceFromRayToPoint(ray, curPoint) <= radius) {
```

Comment:
this may be why it doesn't find the actual closes point....it finds something that's within distance without drilling deeper into the tree.